source/CCustomOpcodeSystem.cpp

```diff
@@ -2406,53 +1980,17 @@ namespace CLEO
 					last_found = index + 1;	// on next opcode call start search from next index
 											//if(last_found >= (unsigned)pool.GetSize()) last_found = 0;
 											// obj.referenceType = 2; // add reference to found actor
-					*thread << pool.GetRef(obj);
-					SetScriptCondResult(thread, true);
-					return OR_CONTINUE;
+
+					auto found = pool.GetRef(obj);
+					OPCODE_WRITE_PARAM_INT(found);
+					OPCODE_CONDITION_RESULT(true);
 				}
 			}
 		}
 
 		last_found = 0;
-		*thread << -1;
-		SetScriptCondResult(thread, false);
-		return OR_CONTINUE;
-	}
-
-	//0AE9=0,pop_float
-	OpcodeResult __stdcall opcode_0AE9(CRunningScript *thread)
-	{
-		float result;
-		_asm fstp result
-		opcodeParams[0].fParam = result;
-		SetScriptParams(thread, 1);
-		return OR_CONTINUE;
-	}
-
-	//0AEA=2,%2d% = actor_struct %1d% handle
-	OpcodeResult __stdcall opcode_0AEA(CRunningScript *thread)
-	{
-		CPed *struc;
-		*thread >> struc;
-		*thread << GetPedPool().GetRef(struc);
-		return OR_CONTINUE;
-	}
-
-	//0AEB=2,%2d% = car_struct %1d% handle
-	OpcodeResult __stdcall opcode_0AEB(CRunningScript *thread)
-	{
-		CVehicle *struc;
-		*thread >> struc;
-		*thread << GetVehiclePool().GetRef(struc);
-		return OR_CONTINUE;
-	}
-
-	//0AEC=2,%2d% = object_struct %1d% handle
-	OpcodeResult __stdcall opcode_0AEC(CRunningScript *thread)
-	{
-		CObject *struc;
-		*thread >> struc;
-		*thread << GetObjectPool().GetRef(struc);
+		OPCODE_WRITE_PARAM_INT(-1);
+		OPCODE_CONDITION_RESULT(false);
 		return OR_CONTINUE;

```

Comment:
I hope all of the above changes are covered with tests. There is no way to catch any error by a naked eye

Reply:
Yeah. Most of memory plugin opcodes has unit tests already created. Filesystem plugin was covered too, but now the test should be moved from single file into file peer opcode scripts.
This would be nice task to invite contributors.